src/rs.cpp

```diff
@@ -4115,9 +4115,9 @@ bool is_pixel_in_line(const float curr[2], const float start[2], const float end
 void adjust_2D_point_to_boundary(float p[2], int width, int height)
 {
     if (p[0] < 0) p[0] = 0;
-    if (p[0] > width) p[0] = (float)width;
+    if (p[0] > width) p[0] = (float)width - 1e-10; // to make sure int(p[0]) + int(p[1])* width stay within [data, data + width * height)
```

Comment:
We had a similar fix from a user that did
`if (p[0] > width - 1) p[0] = (float)width - 1`

Why not doing that?